docs/en/modules/ws2812.md

```diff
@@ -1,57 +1,163 @@
+
+#### Parameters
+none
+
+#### Returns
+`nil`
 
 ## ws2812.write()
-Send GRB data in 8 bits to a WS2812 chain.
+Send data to a led strip using its native format which is generally Green,Red,Blue for RGB strips
+and Green,Red,Blue,White for RGBW strips.
 
 #### Syntax
-`ws2812.writegrb(pin, string)`
+`ws2812.write(string)`
 
 #### Parameters
-- `pin` is ignored. Only works with GPIO2.
-- `string` payload to be sent to one or more WS2812 LEDs.
-  It should be composed from a GRB triplet per element.
-    - `G1` the first pixel's Green channel (0-255)
-    - `R1` the first pixel's Red channel (0-255)
-    - `B1` the first pixel's Blue channel (0-255)<br />
-    ... You can connect a lot of WS2812 ...
-    - `G2`, `R2`, `B2` are the next WS2812's Green, Red, and Blue channel parameters
+- `string` payload to be sent to one or more WS2812 like leds.
 
 #### Returns
 `nil`
 
+#### Example
+```lua
+ws2812.init()
+ws2812.write(string.char(255,0,0,255,0,0) -- turn the two first RGB leds to green
+```
+
 ```lua
-g = 0
-r = 255
-b = 0
-leds_grb = string.char(g,r,b, g,r,b) 
-ws2812.write(2, leds_grb) -- turn two WS2812Bs to red, connected to pin GPIO2
+ws2812.init()
+ws2812.write(string.char(0,0,0,255,0,0,0,255) -- turn the two first RGBW leds to white
 ```
 
-## ws2812.writergb()
-Send GRB data in 8bits to a WS2812 chain.
+# Buffer module
+For more advanced animations, it is useful to keep a "framebuffer" of the strip,
+interact with it and flush it to the strip.
+
+For this purpose, the ws2812 library offers a read/write buffer.
+
+#### Example
+Led chaser with a RGBW strip
+```lua
+local i, b = 0, ws2812.newBuffer(300, 4); b:fill(0,0,0,0); tmr.alarm(0, 50, 1, function()
+        i=i+1
+        b:fade(2)
+        b:set(i%b:size(), 0, 0, 0, 255)
+        b:write()
+end)
+```
+
+## ws2812.newBuffer()
+Allocate a new memory buffer to store led values.
+
+#### Syntax
+`ws2812.newBuffer(numberOfLeds, bytesPerLed)`
+
+#### Parameters
+ - `numberOfLeds` length of the led strip
+ - `bytesPerLed` 3 for RGB strips and 4 for RGBW strips
+
+#### Returns
+`ws2812.buffer`
+
+## ws2812.buffer:get()
+Return the value at the given position
+
+#### Syntax
+`buffer:get(index)`
+
+#### Parameters
+ - `index` position in the buffer
+
+#### Returns
+`(color)`
+
+#### Example
+```lua
+buffer:get(2) -- return the color of the third led
+```
+## ws2812.buffer:set()
+Setthe value at the given position
 
 #### Syntax
-`ws2812.writergb(pin, string)`
+`buffer:set(index, color)`
 
 #### Parameters
-- `pin` is ignored. Only works with GPIO2.
-- `string` payload to be sent to one or more WS2812 LEDs.
-  It should be composed from an RGB triplet per element.
-    - `R1` the first pixel's Red channel (0-255)
-    - `G1` the first pixel's Green channel (0-255)
-    - `B1` the first pixel's Blue channel (0-255)<br />
-    ... You can connect a lot of WS2812 ...
-    - `R2`, `G2`, `B2` are the next WS2812's Red, Green, and Blue channel parameters
+ - `index` position in the buffer
```

Comment:
The range for `index` is 0 to n, but the [Lua convention for arrays](https://www.lua.org/pil/11.1.html) is to start at index 1. Should this be applied here as well (and for `ws2812.buffer:get()`)?